test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -1290,4 +1310,101 @@ TEST_P(ExtProcIntegrationTest, BufferBodyOverridePostWithRequestBody) {
   verifyDownstreamResponse(*response, 200);
 }
 
+// Set up per-route configuration that sets a custom processing mode on the
+// route, and test that the processing mode takes effect.
+TEST_P(ExtProcIntegrationTest, PerRouteProcessingMode) {
+  initializeConfig();
+  config_helper_.addConfigModifier([this](HttpConnectionManager& cm) {
+    // Set up "/foo" so that it will send a buffered body
+    auto* vh = cm.mutable_route_config()->mutable_virtual_hosts()->Mutable(0);
+    auto* route = vh->mutable_routes()->Mutable(0);
+    route->mutable_match()->set_path("/foo");
+    ExtProcPerRoute per_route;
+    per_route.mutable_overrides()->mutable_processing_mode()->set_response_body_mode(
+        ProcessingMode::BUFFERED);
+    setPerRouteConfig(route, per_route);
+  });
+  HttpIntegrationTest::initialize();
+
+  auto response =
+      sendDownstreamRequest([](Http::RequestHeaderMap& headers) { headers.setPath("/foo"); });
+  processRequestHeadersMessage(true, absl::nullopt);
+  Buffer::OwnedImpl full_response;
+  TestUtility::feedBufferWithRandomCharacters(full_response, 100);
+  handleUpstreamRequestWithResponse(full_response, 100);
+  processResponseHeadersMessage(false, absl::nullopt);
+  // Because of the per-route config we should get a buffered response
+  processResponseBodyMessage(false, [&full_response](const HttpBody& body, BodyResponse&) {
+    EXPECT_TRUE(body.end_of_stream());
+    EXPECT_EQ(body.body(), full_response.toString());
+    return true;
+  });
+  verifyDownstreamResponse(*response, 200);
+}
+
+// Set up configuration on the virtual host and on the route and see that
+// the two are merged.
```

Comment:
Is this how the other per route configurations work? I may be misremembering but I thought most they wouldn't merge, but pick the most specific one

Reply:
I find this confusing myself. I copied the language in the .proto from the ext_authz proto, which says, "If disabled is specified in multiple per-filter-configs, the most specific one will be used."

I think that this is the right logic for this. In the future, we'll implement overrides for request and response properties, and in that case I think that we might want to try and merge them if that makes sense.

Reply:
So for the two properties that are supported today, the behavior is indeed to pick the most specific one -- but we might choose something different for the properties that aren't supported yet.

Reply:
After chatting a bit with other maintainers I think this is fine, as the behavior appears to be documented.